tests/test_observability/conftest.py

```
@@ -0,0 +1,144 @@
+
+    Use when you need to capture and inspect spans created during tests.
+    Returns a tuple of (provider, exporter) for easy span inspection.
+    """
+    exporter = InMemorySpanExporter()
+    provider = TracerProvider()
+    provider.add_span_processor(SimpleSpanProcessor(exporter))
+    return provider, exporter
+
+
+@pytest.fixture
+def captured_spans(mock_tracer_provider, reset_tracing):
```

Ruff (0.13.3)

Unused function argument: `reset_tracing`

(ARG001)